libraries/GCS_MAVLink/GCS_Common.cpp

```diff
@@ -761,6 +761,33 @@ void GCS_MAVLINK::handle_radio_status(mavlink_message_t *msg, DataFlash_Class &d
     }
 }
 
+/*
+  return a timesync request
+  Sends back ts1 as received, and tc1 is the local timestamp in usec
+ */
+void GCS_MAVLINK::handle_timesync(mavlink_message_t *msg)
+{
+    // Decode incoming timesync message
+    mavlink_timesync_t tsync;
+    mavlink_msg_timesync_decode(msg, &tsync);
+
+    // Create new timestruct to return
+    mavlink_timesync_t rsync;
+    rsync.ts1 = tsync.ts1;
+    rsync.tc1 = AP_HAL::micros64();
```

Comment:
Not normally a time guy, but wouldn't the best numbers for timing come from capturing the micros64() number as soon as you enter the function, given that you call it in 2 places? (I get that this is a tiny tiny tiny tiny amount of jitter but stil).